src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,56 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exeptions.UserRegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        boolean checkRegistrationAllowance = checkLogin(user.getLogin())
+                && checkPassword(user.getPassword())
+                && checkAge(user.getAge())
+                && !userExists(user);
+        if (!checkRegistrationAllowance) {
+            throw new UserRegistrationException("Registration failed ");
+        }
+        return storageDao.add(user);
+    }
+
+    private boolean checkPassword(String password) {
+        if (password == null || password.length() < MIN_PASSWORD_LENGTH) {
+            throw new UserRegistrationException("Invalid password , password shoudnt be less than "
+                    + MIN_PASSWORD_LENGTH + " characters");
+        }
```

Comment:
separate the null check from it, because right now if it is null you just say that user has to be of valid age, and nothing about null, which is misguiding.
Same for other validations